tfhe/src/typed_api/booleans/types/base.rs

```diff
@@ -0,0 +1,313 @@
+use std::borrow::Borrow;
+use std::ops::{BitAnd, BitOr, BitXor};
+
+use crate::boolean::ciphertext::{Ciphertext, CompressedCiphertext};
+use serde::{Deserialize, Serialize};
+
+use crate::typed_api::booleans::client_key::GenericBoolClientKey;
+use crate::typed_api::booleans::public_key::GenericBoolPublicKey;
+use crate::typed_api::booleans::parameters::BooleanParameterSet;
+use crate::typed_api::booleans::server_key::GenericBoolServerKey;
+use crate::typed_api::global_state::WithGlobalKey;
+use crate::typed_api::keys::{ClientKey, PublicKey, RefKeyFromKeyChain, RefKeyFromPublicKeyChain};
+use crate::typed_api::traits::{FheDecrypt, FheEncrypt, FheEq, FheTryEncrypt, FheTryTrivialEncrypt, FheTrivialEncrypt};
+
+/// The FHE boolean data type.
+///
+/// To be able to use this type, the cargo feature `booleans` must be enabled,
+/// and your config should also enable the type with either default parameters or custom ones.
+///
+/// # Example
+/// ```rust
+/// use tfhe::prelude::*;
+/// #[cfg(feature = "boolean")]
+/// # {
+/// use tfhe::{generate_keys, set_server_key, ConfigBuilder, FheBool};
+///
+/// // Enable booleans in the config
+/// let config = ConfigBuilder::all_disabled().enable_default_bool().build();
+///
+/// // With the booleans enabled in the config, the needed keys and details
+/// // can be taken care of.
+/// let (client_key, server_key) = generate_keys(config);
+///
+/// let ttrue = FheBool::encrypt(true, &client_key);
+/// let ffalse = FheBool::encrypt(false, &client_key);
+///
+/// // Do not forget to set the server key before doing any computation
+/// set_server_key(server_key);
+///
+/// let fhe_result = ttrue & ffalse;
+///
+/// let clear_result = fhe_result.decrypt(&client_key);
+/// assert_eq!(clear_result, false);
+/// # }
+/// ```
+// #[cfg_attr(doc, cfg(feature = "boolean"))]
```

Comment:
this looks like a comment ?

also I think I saw Sarah use a docrs attribute in some instances to indicate by which feature some things are gated, we could maybe add that at the lib/mod.rs level (not necessary for this PR) just a thought that poped while reading this

Reply:
I don't know what Sarah used, but this is not supposed to be commented out,
when not commented out and when building the doc, it shows a label on the generator html that tells which features need to be activated.

eg: https://docs.rs/concrete/0.2.1/concrete/#structs

Reply:
I think she uses a docrs attribute